wagtail/admin/views/generic/models.py

```diff
@@ -327,17 +319,48 @@ def get_add_url(self):
             return reverse(self.add_url_name)
 
     def get_valid_orderings(self):
-        orderings = []
+        orderings = set()
+        # Gather valid orderings from the Column instances
         for col in self.columns:
             if col.sort_key:
-                orderings.append(col.sort_key)
-                orderings.append("-%s" % col.sort_key)
+                orderings |= {col.sort_key, f"-{col.sort_key}"}
+
+        # Combine with any explicitly defined orderings
+        default_ordering = self.ordering or self.default_ordering
+        if default_ordering:
+            if isinstance(default_ordering, str):
+                default_ordering = (default_ordering,)
+            orderings |= set(default_ordering)
         return orderings
 
     def get_ordering(self):
-        ordering = self.request.GET.get("ordering", self.default_ordering)
+        default_ordering = self.ordering or self.default_ordering
```

Comment:
I don't know why we have a default_ordering attribute on the generic IndexView. It extends Django's BaseListView, so it has the ordering attribute. Why not use it? 🤔